src/utils/transaction/userOperation.ts

```diff
@@ -43,12 +45,26 @@ export const generateSignedUserOperation = async (
     address: accountAddress as Address,
   });
 
-  // Resolve Factory Contract Address from Smart-Account Contract
-  const accountFactoryAddress = await readContract({
-    contract: smartAccountContract,
-    method: "function factory() view returns (address)",
-    params: [],
-  });
+  // use the user provided factory address if available
+  let accountFactoryAddress = userProvidedAccountFactoryAddress;
+
+  if (!accountFactoryAddress) {
+    // Resolve Factory Contract Address from Smart-Account Contract
+    try {
+      const onchainAccountFactoryAddress = await readContract({
+        contract: smartAccountContract,
+        method: "function factory() view returns (address)",
+        params: [],
+      });
+
+      accountFactoryAddress = getAddress(onchainAccountFactoryAddress);
+    } catch (e) {
+      // if no factory address is found, throw an error
+      throw new Error(
+        `Failed to find factory address for account '${accountAddress}' on chain '${chainId}'`,
+      );
+    }
+  }
```

Comment:
Not sure if this `catch` is necessary unless the caught exception is hard to parse. Usually I try to avoid catching + rethrowing unless you explicitly want to wrap the error message.

Reply:
I wanted us to explicitly know from logs that get `factoryAddress` step went wrong, otherwise it's a read contract call error